feature/experimental/bgp/ate_tests/bgp_remove_private_as/bgp_remove_private_as_test.go

```diff
@@ -0,0 +1,350 @@
+	if gotInstalled := gnmi.Get(t, dut, prefixesv4.Installed().State()); gotInstalled != wantInstalled {
+		t.Errorf("Installed prefixes mismatch: got %v, want %v", gotInstalled, wantInstalled)
+	}
+	if gotSent := gnmi.Get(t, dut, prefixesv4.Sent().State()); gotSent != wantSent {
+		t.Errorf("Sent prefixes mismatch: got %v, want %v", gotSent, wantSent)
+	}
+}
+
+// configureATE configures the interfaces and BGP protocols on an ATE, including
+// advertising some(faked) networks over BGP.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice, asSeg []uint32, asMode string) *ondatra.ATETopology {
+	port1 := ate.Port(t, "port1")
+	topo := ate.Topology().New()
+	iDut1 := topo.AddInterface(ateSrc.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateSrc.IPv4CIDR()).WithDefaultGateway(dutSrc.IPv4)
+
+	port2 := ate.Port(t, "port2")
+	iDut2 := topo.AddInterface(ateDst.Name).WithPort(port2)
+	iDut2.IPv4().WithAddress(ateDst.IPv4CIDR()).WithDefaultGateway(dutDst.IPv4)
+
+	// Setup ATE BGP route v4 advertisement.
+	bgpDut1 := iDut1.BGP()
+	bgpDut1.AddPeer().WithPeerAddress(dutSrc.IPv4).WithLocalASN(ateAS1).
+		WithTypeExternal()
+
+	bgpDut2 := iDut2.BGP()
+	bgpDut2.AddPeer().WithPeerAddress(dutDst.IPv4).WithLocalASN(ateAS2).
+		WithTypeExternal()
+
+	bgpNeti1 := iDut1.AddNetwork("bgpNeti1")
+	bgpNeti1.IPv4().WithAddress(advertisedRoutesv4CIDR).WithCount(routeCount)
+	bgpNeti1.BGP().WithNextHopAddress(ateSrc.IPv4)
+
+	if asMode == "SEQ" {
+		bgpNeti1.BGP().AddASPathSegment(asSeg...).WithTypeSEQ()
+	} else {
+		// TODO : SET mode is not working
+		// https://github.com/openconfig/featureprofiles/issues/1659
+		bgpNeti1.BGP().AddASPathSegment(asSeg...).WithTypeSET()
+	}
+
+	t.Logf("Pushing config to ATE and starting protocols...")
+	topo.Push(t)
+	topo.StartProtocols(t)
+
+	return topo
+}
+
+type bgpNeighbor struct {
+	as         uint32
+	neighborip string
+	isV4       bool
+	peerGrp    string
+}
+
+// verifyBGPAsPath is to Validate AS Path attribute using bgp rib telemetry on ATE.
+func verifyBGPAsPath(t *testing.T, dut *ondatra.DUTDevice, ate *ondatra.ATEDevice, asSeg []uint32, removeASPath bool) {
+	at := gnmi.OC()
+	rib := at.NetworkInstance(ateDst.Name).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "0").Bgp().Rib()
+	prefixPath := rib.AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Ipv4Unicast().
+		NeighborAny().AdjRibInPre().RouteAny().WithPathId(0).Prefix()
+
+	gnmi.WatchAll(t, ate, prefixPath.State(), time.Minute, func(v *ygnmi.Value[string]) bool {
+		_, present := v.Val()
+		return present
+	}).Await(t)
+
+	var wantAsSeg = []uint32{dutAS, ateAS1}
+	if !removeASPath {
+		wantAsSeg = append(wantAsSeg, asSeg...)
+	} else {
+		for _, as := range asSeg {
+			if as < 64512 {
+				wantAsSeg = append(wantAsSeg, as)
+			}
+		}
+	}
+	var gotAsSeg []uint32
+	_, ok := gnmi.WatchAll(t, ate, rib.AttrSetAny().AsSegmentAny().State(), 1*time.Minute, func(v *ygnmi.Value[*oc.NetworkInstance_Protocol_Bgp_Rib_AttrSet_AsSegment]) bool {
```

Comment:
you should be able to make this:
`gotASSeg, ok := gnmi.WatchAll(...`
and remove the var declaration just above and the "gotAsSeg =" line from inside the WatchAll

Reply:
updated. thank you